scripts/clean_osm_data.py

```diff
@@ -835,6 +835,65 @@ def set_name_by_closestcity(df_all_generators, colname="name"):
     return df_all_generators
 
 
+def use_custom_data_files(custom_component_type):
+    """
+    Function to check if OSM or custom data should be considered.
+
+    the custom_component_type should be a string named lines, cables or
+    substations
+    """
+
+    # checks the type of data/components to be reviewed
+    if custom_component_type == "lines":
+        custom_conditional = snakemake.config["clean_osm_data_options"][
+            "use_custom_lines"
+        ]
+        custom_path = snakemake.config["clean_osm_data_options"].get(
+            "path_custom_lines", False
+        )
+
+    elif custom_component_type == "substations":
+        custom_conditional = snakemake.config["clean_osm_data_options"][
+            "use_custom_substations"
+        ]
+        custom_path = snakemake.config["clean_osm_data_options"].get(
+            "path_custom_substations", False
+        )
+
+    elif custom_component_type == "cables":
+        custom_conditional = snakemake.config["clean_osm_data_options"][
+            "use_custom_cables"
+        ]
+        custom_path = snakemake.config["clean_osm_data_options"].get(
+            "path_custom_cables", False
+        )
+
+    else:
+        raise ValueError(
+            "Invalid custom_component_type. Expected 'lines', 'substations', or 'cables'."
+        )
```

Comment:
This code can be revised to drop the if-elif case, such as
```suggestion
     try:
         custom_opt = clean_options[f"use_custom_{network_asset}"]
         custom_path = clean_options[f"path_custom_{network_asset}"]
     catch:
         logger.error(f"Missing use_custom_{network_asset} or path_custom_{network_asset} options in the config file")
```

May you test if this code works but keep the functionality?